templates/geosafe/analysis/dynamic_scripts/create/transparency_control.html

```diff
@@ -152,6 +152,31 @@
                 $("#impact-legend-img").attr("src", legend_url);
                 impact_layer.setZIndex(3);
             }
+
+            // Zooming magic
+            // If only one of the 3 layers is selected then zoom to the layer
+            if ((is_hazard != is_exposure) != is_impact) {
+              map.fitBounds([[layer_bbox_y1, layer_bbox_x1], [layer_bbox_y0, layer_bbox_x0]]);
+            }
+            // If hazard and exposure selected intersect both extents
+            if ((layer_purpose=='hazard' && is_exposure) || (layer_purpose=='exposure' && is_hazard)) {
```

Comment:
I might be wrong.
but, i think the correct statement might be:
```if ((layer_purpose=='hazard' && is_hazard) && (layer_purpose=='exposure' && is_exposure))```
the `layer_purpose == hazard` should go with `is_hazard`, while `layer_purpose== exposure` go with `is_exposure`.
also, since you want to check whether both layers are selected,  should the relation use `&&`, instead of `||` ?